test/e2e_federation/ingress.go

```diff
@@ -322,8 +322,9 @@ func verifyCascadingDeletionForIngress(clientset *fedclientset.Clientset, cluste
 func waitForIngressOrFail(clientset *kubeclientset.Clientset, namespace string, ingress *v1beta1.Ingress, present bool, timeout time.Duration) {
 	By(fmt.Sprintf("Fetching a federated ingress shard of ingress %q in namespace %q from cluster", ingress.Name, namespace))
 	var clusterIngress *v1beta1.Ingress
-	err := wait.PollImmediate(framework.Poll, timeout, func() (bool, error) {
-		clusterIngress, err := clientset.Ingresses(namespace).Get(ingress.Name, metav1.GetOptions{})
+	var err error
+	err = wait.PollImmediate(framework.Poll, timeout, func() (bool, error) {
+		clusterIngress, err = clientset.Ingresses(namespace).Get(ingress.Name, metav1.GetOptions{})
```

Comment:
Just put the `var err error` inside the func() closure. even though err is same name, they could be 2 different variables.